components/webgpu/ipc_messages/send_dom.rs

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+//! IPC massages that are send to GPU DOM objects.
```

Comment:
Are the DOM objects receiving these messages directly or are they somehow forwarded through the script thread?

If this is the later case:

```suggestion
//! IPC messages that are sent to the script thread and forwarded to WebGPU DOM objects.
```

Or the former:

```suggestion
//! IPC messages that are sent to the receivers on WebGPU DOM objects.
```

Reply:
I think technically this is script thread (or maybe subthread of script thread, page scope; I am not so really confident in my understanding of threading in script), but what I want to say is that they are revived using https://github.com/servo/servo/blob/f29dd64a7b633e844756e6eecf9e05e0b327fc51/components/script/dom/gpu.rs#L47 which is impl on Dom objects that receives it (underneath this works by creating dom task source ...)

Then there are messages that are actually send to script_thread: https://github.com/servo/servo/blob/f29dd64a7b633e844756e6eecf9e05e0b327fc51/components/script/script_thread.rs#L2422